packages/experimental/kai/src/app/pages/RecoverIdentityPage.tsx

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import { SingleInputStep } from '@dxos/react-appkit';
+import { useClient, useIdentity } from '@dxos/react-client';
+import { Heading, useTranslation } from '@dxos/react-components';
+
+import { useOptions } from '../../hooks';
+import { Generator } from '../../proto';
+
+// NOTE: Copied from halo-app.
+// TODO(wittjosiah): Utilize @dxos/react-ui patterns.
+
+export const RecoverIdentityPage = () => {
```

Comment:
Is there a goal to try to factor out these "pages" into patterns (without assuming particular routes?)

Reply:
In general yes but trying to avoid prematurely abstracting things out. We do expect the auth flows to get factored out into the os patterns but not clear yet what the shape of those pieces that get factored out are.

In this case, it's just the first time this is being re-used so makes more sense to copy it and see how things shape up.